core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -151,7 +151,15 @@ void start() {
       persistedCaches.forEach((name, configuration) -> {
          ensurePersistenceCompatibility(name, configuration);
          // The cache configuration was permanent, it still needs to be
-         CompletionStages.join(getOrCreateCache(name, configuration, adminFlags));
+         CompletionStages.join(createCacheInternal(name, null, configuration, adminFlags)
+               .thenCompose(r -> {
+                  if (r instanceof CacheState) {
```

Comment:
This is because of a failure that shows up with the `GracefulShutdownRestartIT` test. The test does some operations, restart two nodes, and does some operations again.

There are times during the restart this step of creating from persisted configuration fails. The `putIfAbsent` operation fails as the other node already has the configuration in the config cache, and the ConfigurationListener is not notified, the cache does not start, and afterward, it always fails because the configuration does not exist locally.

Now we try to create from persistence, and if there is already a configuration remotely, we verify compatibility and use THE REMOTE. This is during the start, so if we're talking about a case of concurrency, it is OK, as both nodes have the same configuration. But if we're talking about a node joining later, the persisted configuration could be outdated.